src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -452,23 +453,23 @@ public async Task<bool> TryRefreshAsync(CancellationToken cancellationToken)
             {
                 if (IsAuthenticationError(e))
                 {
-                    _logger?.LogWarning(e, LoggingConstants.RefreshFailedDueToAuthenticationError);
+                    _logger.LogWarning(LoggingConstants.RefreshFailedDueToAuthenticationError, e);
                 }
                 else
                 {
-                    _logger?.LogWarning(e, LoggingConstants.RefreshFailedError);
+                    _logger.LogWarning(LoggingConstants.RefreshFailedError, e);
                 }
 
                 return false;
             }
             catch (KeyVaultReferenceException e)
             {
-                _logger?.LogWarning(e, LoggingConstants.RefreshFailedDueToKeyVaultError);
+                _logger.LogWarning(LoggingConstants.RefreshFailedDueToKeyVaultError, e);
                 return false;
             }
             catch (OperationCanceledException)
             {
-                _logger?.LogWarning(LoggingConstants.RefreshCanceledError);
+                _logger.LogWarning(LoggingConstants.RefreshCanceledError, null);

```

Comment:
If we have to pass null for an argument, there is scope for improving that method. 
In this case, we can change LogWarning method to accept just the final message and pre-build the message from exception before calling LogWarning.